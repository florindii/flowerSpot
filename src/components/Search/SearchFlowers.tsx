import React, { ChangeEvent } from 'react';
import searchBackgroundImage from '../../assets/icons/pl-icon-search.svg';
import './Search.scss';

interface SearchFlowersProps {
  searchTerm: string;
  setSearchTerm: (term: string) => void;
}

const SearchFlowers: React.FC<SearchFlowersProps> = ({ searchTerm, setSearchTerm }) => {
  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(e.target.value);
  };

  return (
    <div className='search-holder'>
        <div className="search-text-holder">
            <h2 className='discover-flowers'>Discover flowers around you</h2>
            <p className='explore-flowers'>Explore between more than 8.427 sightings</p>
        </div>
        <div className='search-input'>
        <input
            className='form-control'
            type="text"
            placeholder="Looking for something specific?"
            value={searchTerm}
            onChange={handleSearchChange}
        />
        <img src={searchBackgroundImage} alt="search" />
        </div>
    </div>
  );
};

export default SearchFlowers;
