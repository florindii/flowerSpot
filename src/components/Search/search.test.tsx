import React from 'react';
import renderer from 'react-test-renderer';
import SearchFlowers from './SearchFlowers';

describe('SearchFlowers Component', () => {
  it('renders correctly with provided props', () => {
    const searchTerm = 'Flowers';
    const setSearchTerm = jest.fn();

    const tree = renderer
      .create(<SearchFlowers searchTerm={searchTerm} setSearchTerm={setSearchTerm} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('triggers setSearchTerm function on input change', () => {
    const searchTerm = 'Flowers';
    const setSearchTerm = jest.fn();

    const component = renderer.create(
      <SearchFlowers searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
    );

    // Find the input element and simulate a change event
    const inputElement = component.root.findByType('input');
    inputElement.props.onChange({ target: { value: 'New Search Term' } });

    // Check if setSearchTerm function was called with the new value
    expect(setSearchTerm).toHaveBeenCalledWith('New Search Term');
  });
});
