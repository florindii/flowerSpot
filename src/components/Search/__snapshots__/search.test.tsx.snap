// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFlowers Component renders correctly with provided props 1`] = `
<div
  className="search-holder"
  id="search"
>
  <div
    className="search-text-holder"
  >
    <h2
      className="discover-flowers"
    >
      Discover flowers around you
    </h2>
    <p
      className="explore-flowers"
    >
      Explore between more than 8.427 sightings
    </p>
  </div>
  <div
    className="search-input"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Looking for something specific?"
      type="text"
      value="Flowers"
    />
    <img
      alt="search"
      src="pl-icon-search.svg"
    />
  </div>
</div>
`;
